File: benchmarks/system/list_ops.py
"""Benchmark: Prime sieve — Python equivalent."""

def is_prime(n):
    if n < 2: return False
    if n < 4: return True
    if n % 2 == 0: return False
    d = 3
    while d * d <= n:
        if n % d == 0: return False
        d += 2
    return True

def main():
    count = 0
    for i in range(100_000):
        if is_prime(i):
            count += 1
    print(f"primes = {count}")

if __name__ == "__main__":
    main()

File: benchmarks/system/list_ops.rs
use std::time::Instant;
// Benchmark: Prime sieve — Rust equivalent.
fn is_prime(n: i64) -> bool {
    if n < 2 { return false; }
    if n < 4 { return true; }
    if n % 2 == 0 { return false; }
    let mut d = 3i64;
    while d * d <= n {
        if n % d == 0 { return false; }
        d += 2;
    }
    true
}

fn main() {
    let __bench_t0 = Instant::now();

    let mut count = 0i64;
    for i in 0..100_000i64 {
        if is_prime(i) {
            count += 1;
        }
    }
    println!("primes = {}", count);
    let __bench_dt = __bench_t0.elapsed().as_secs_f64();
    println!("__BENCH_METRICS__");
    println!("wall_time_s={}", __bench_dt);
    println!("cpu_time_s={}", __bench_dt);
    println!("peak_memory_kb=0");
}
